File: 08-DesignPatterns-Java/Solution/Strategy/src/Ducks/RubberDuck.java
package Ducks;

import Fly.NoneFlyStrategy;
import Quack.SqueakQuackStrategy;

public class RubberDuck extends Duck {
    public RubberDuck() {
        super(new SqueakQuackStrategy(), new NoneFlyStrategy());
    }

    @Override
    public void display() {
        System.out.println("All yellow, made from rubber.");
    }
}

File: 09-DesignPatterns-CSharp/09-1-3-Strategy-Solution/Ducks/RubberDuck.cs
namespace Strategy.Ducks
{
    using System;
    using Strategy.Fly;
    using Strategy.Quack;

    public sealed class RubberDuck : Duck
    {
        public RubberDuck()
            : base(new SqueakQuackStrategy(), new NoneFlyStrategy())
        {
        }

        public override void Display() => Console.WriteLine("All yellow, made from rubber.");
    }
}